src/mutations/item.ts

```diff
@@ -125,7 +126,19 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
             return old.set(idx, newItem);
           },
         }),
-        item: await mutateItem({ id: newItem.id, value: newItem }),
+        item: await (async () => {
+          const itemKey = buildItemKey(newItem.id);
+          await queryClient.cancelQueries(itemKey);
+          const prevValue = queryClient.getQueryData(itemKey) as Record<Item>;
+          queryClient.setQueryData(
+            itemKey,
+            Map({
```

Comment:
Is there no way of doing this with Immutable? I think you should be able to replace keys.

Reply:
You're right, `.merge` is exactly what I needed.